src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.draftstore.config;
+
+import org.springframework.boot.actuate.trace.TraceProperties;
+import org.springframework.boot.actuate.trace.TraceRepository;
+import org.springframework.boot.actuate.trace.WebRequestTraceFilter;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toSet;
+import static uk.gov.hmcts.reform.draftstore.service.AuthService.SECRET_HEADER;
+import static uk.gov.hmcts.reform.draftstore.service.AuthService.SERVICE_HEADER;
+
+@Component
+public class RequestTraceFilter extends WebRequestTraceFilter {
+
+    public final static List<String> sensitiveHeaders =
+        Arrays.asList(
+            SECRET_HEADER,
+            SERVICE_HEADER
+        );
+
+    public RequestTraceFilter(TraceRepository repository, TraceProperties properties) {
+        super(repository, properties);
+    }
+
+    @Override
+    protected void postProcessRequestHeaders(Map<String, Object> headers) {
+        headers
+            .keySet()
+            .stream()
+            .filter(header -> shouldBeRemoved(header))
+            .collect(toSet())
+            .forEach(h -> headers.remove(h));
```

Comment:
```java
            .filter(this::shouldBeRemoved)
            .collect(toSet())
            .forEach(headers::remove);
```

Reply:
Personally, I like lambdas more, as they provide slightly more information (by naming the parameter), which makes it easier to figure out what's going on. Probably a matter of preference, though.